cmake/OpenCVFindLAPACK.cmake

```diff
@@ -107,17 +130,11 @@ macro(ocv_lapack_check)
     endif()
 
     set(LAPACK_TRY_COMPILE_DEF "")
-    if(LAPACK_IMPL STREQUAL "LAPACK/Apple" AND NOT IOS) # https://github.com/opencv/opencv/issues/24660
-      # Get macOS version
-      execute_process(COMMAND sw_vers -productVersion
-                      OUTPUT_VARIABLE MACOS_VERSION
-                      OUTPUT_STRIP_TRAILING_WHITESPACE)
-      # Enable Accelerate New LAPACK if macOS >= 13.3
-      if (MACOS_VERSION VERSION_GREATER "13.3" OR MACOS_VERSION VERSION_EQUAL "13.3")
-        set(LAPACK_TRY_COMPILE_DEF "-DACCELERATE_NEW_LAPACK")
-        add_compile_definitions(ACCELERATE_NEW_LAPACK)
-        add_compile_definitions(ACCELERATE_LAPACK_ILP64)
-      endif()
+    if(LAPACK_IMPL STREQUAL "LAPACK/Apple" AND OPENCV_OSX_USE_ACCELERATE_NEW_LAPACK)
+      message(STATUS "LAPACK(${LAPACK_IMPL}): Accelerate New LAPACK is enabled.")
+      set(LAPACK_TRY_COMPILE_DEF "-DACCELERATE_NEW_LAPACK")
```

Comment:
I don't think it's necessary to try-compile this, but I guess it's not harmful.

Reply:
Just to keep aligned with what we are doing.